assembl/static/js/app/views/messageListPostQuery.js

```diff
@@ -374,6 +389,33 @@ var PostQuery = function() {
       });
     };
 
+    this.getResultMessageStructureCollectionPromise = function() {
+      // The "deleted messages" filter needs a "all message structure collection" which is different from the default one (because the default one may not include all deleted messages)
+      // Maybe we should use a Subset of the getAllMessageStructureCollectionPromise instead
+      // TODO: Have a more generic mecanism than a list of ifs. For example add a usesNonStandardMessageStructure boolean method in AbstractPostFilter
+      if ( ("only_deleted_posts" in this._query) || ("also_deleted_posts" in this._query) ){
```

Comment:
I would highly recommend making an `enum` type object that contains these strings. Will keep the code DRY, and we can document the strings wherever they are